test/jdk/javax/swing/JTable/JTableRightOrientationTest.java

```diff
@@ -0,0 +1,210 @@
+                    frame.pack();
+                    frame.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+                    frame.applyComponentOrientation(ComponentOrientation.RIGHT_TO_LEFT);
+                    frame.setVisible(true);
+                });
+
+                robot.waitForIdle();
+                robot.delay(1000);
+                SwingUtilities.invokeAndWait(() -> {
+                    allColumnWidths = 0;
+                    for (int i = 0; i < customTableObj.table.getColumnCount(); i++) {
+                        allColumnWidths += customTableObj.table.getTableHeader().getColumnModel()
+                                .getColumn(i)
+                                .getWidth();
+                    }
+                    tableLocation = customTableObj.table.getLocationOnScreen();
+                    tableSize = customTableObj.table.getSize();
+                });
+
+                BufferedImage bufferedImage = robot.createScreenCapture(
+                        new Rectangle(tableLocation.x,
+                                tableLocation.y,
+                                (int)tableSize.getWidth() - allColumnWidths,
+                                (int)tableSize.getHeight()));
+                int expectedRGB = bufferedImage.getRGB(0, 0);
+                for (int x = 0; x < bufferedImage.getWidth(); x++) {
+                    for (int y = 0; y < bufferedImage.getHeight(); y++) {
+                        if (expectedRGB != bufferedImage.getRGB(x, y)) {
+                            saveImage(bufferedImage, "failureImage.png");
+                            throw new RuntimeException("Test Failed at <" + x + ", " + y + ">");
+                        }
+                    }
+                }
+
+                System.out.println("Test Passed!");
+            } finally {
+                SwingUtilities.invokeAndWait(() -> {
+                    if (frame != null) {
+                        frame.dispose();
+                    }
+                });
+            }
+            robot.waitForIdle();
+            robot.delay(200);
+        }
+    }
+
+    private static void saveImage(BufferedImage image, String fileName) {
+        try {
+            ImageIO.write(image, "png", new File(fileName));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static void setLookAndFeel(UIManager.LookAndFeelInfo laf) {
+        try {
+            UIManager.setLookAndFeel(laf.getClassName());
+        } catch (UnsupportedLookAndFeelException ignored) {
+            System.out.println("Unsupported LAF: " + laf.getClassName());
+        } catch (ClassNotFoundException | InstantiationException
+                 | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+}
+
+class CustomTable {
+    public final static int COL_FIRSTNAME = 0;
+    public final static int COL_LASTNAME = 1;
+    public final static int COL_SALARY = 2;
```

Comment:
```suggestion
    private static final int COL_FIRSTNAME = 0;
    private static final int COL_LASTNAME = 1;
    private static final int COL_SALARY = 2;
```
Let's use the blessed modifier order; these can be `private`.